pkg/api/customization/cluster/validator.go

```diff
@@ -49,3 +70,80 @@ func (v *Validator) validateLocalClusterAuthEndpoint(request *types.APIContext,
 
 	return nil
 }
+
+func (v *Validator) validateEnforcement(request *types.APIContext, spec *v3.ClusterSpec) error {
+	//if not rke, return
+	if spec.RancherKubernetesEngineConfig == nil {
+		return nil
+	}
+
+	ma := gaccess.MemberAccess{
+		Users:     v.Users,
+		GrLister:  v.GrLister,
+		GrbLister: v.GrbLister,
+	}
+
+	//if user is admin or belongs to whitelist roles, no checks needed
+	callerID := request.Request.Header.Get(gaccess.ImpersonateUserHeader)
+
+	isAdmin, err := ma.IsAdmin(callerID)
+	if err != nil {
+		return err
+	}
+	if isAdmin {
+		return nil
+	}
+	/*if user to whitelist roles, no checks needed
+	whitelisted_roles := settings.EnforcementWhitelistRoles.Get()
+	*/
+
+	if !strings.EqualFold(settings.ClusterTemplateEnforcement.Get(), "true") {
+		return nil
+	}
+	//enforcement is true, template is a must
+	if spec.ClusterTemplateName == "" || spec.ClusterTemplateRevisionName == "" {
+		return httperror.NewFieldAPIError(httperror.MissingRequired, "", "A clusterTemplate is required to create a cluster")
+	}
+
+	enforced, err := v.isTemplateEnforced(spec)
+	if err != nil {
+		return err
+	}
+
+	if !enforced {
+		return httperror.NewFieldAPIError(httperror.PermissionDenied, "", "This is not a valid clusterTemplate to create a cluster")
+	}
+
+	return nil
+}
+
+func (v *Validator) isTemplateEnforced(spec *v3.ClusterSpec) (bool, error) {
+	//template must be marked as enforced=true  + must be public
+
+	clusterTemp, err := v.ClusterTemplateLister.Get(namespace.GlobalNamespace, spec.ClusterTemplateName)
+	if err != nil {
+		return false, err
+	}
+	if !clusterTemp.Spec.Enforced {
+		return false, nil
+	}
+
+	//make sure it is public
```

Comment:
can we just check the member attribute on the template? Would save us on the API call :)